srcCppCommon/PerftestPrinter.cxx

```diff
@@ -0,0 +1,238 @@
+
+
+/*
+ * (c) 2005-2020  Copyright, Real-Time Innovations, Inc. All rights reserved.
+ * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
+ */
+
+
+#include "PerftestPrinter.h"
+
+PerftestOuputFormat get_output_format(std::string outputFormat)
+{
+    if(outputFormat == "legacy"){
+        return LEGACY;
+    } else {
+        return REGULAR;
+    }
+}
+
+PerftestPrinter::PerftestPrinter()
+{
+    _dataLength = 100;
+    _outputFormat = REGULAR;
+    _printSummaryHeaders = true;
+}
+
+void PerftestPrinter::set_header_printed(bool headerPrinted)
+{
+    _headerPrinted = headerPrinted;
+}
+
+void PerftestPrinter::set_print_invertals(bool printIntervals)
+{
+    _printIntervals = printIntervals;
+}
+
+void PerftestPrinter::set_output_format(std::string outputFormat)
+{
+    _outputFormat = get_output_format(outputFormat);
+}
+
+void PerftestPrinter::set_data_length(unsigned int dataLength)
+{
+    if (_outputFormat == LEGACY){
+        printf("\n\n********** New data length is %d\n",
+                dataLength);
+    } else {
+    _dataLength = dataLength;
```

Comment:
I believe in the text I mentioned I would write (depending on wether this is a thoughput test or a latency test):

Intervals One-way Latency for XXX Bytes:

or

Intervals Throughput for XXX Bytes:

Reply:
You will also add here an if to check if we print the headers of we don't, and don't print it if we don't want the headers.

Reply:
Actually, I would divide this function. I will explain you further in a call, but basically I suggest we do in the place where you are calling this function (in perftest_cpp):


```
set_data_length(dataLength);
if (printIntervals) {
   print_latency_interval_header();
}
```

Reply:
ok, after discussing this with @pparrilla:

- In the perftest_cpp, we would do something like:

```
if (printIntervals) {
    new_datalen(dataLen);
}
```

And that function would do something like:

```
_dataLength = _dataLen;

if (printHeaders) {
   print_latency_interval_header();
}

```

Then print_latency_interval_header() would print:

```
switch(_outputFormat)
case LEGACY:
printf("\n\n********** New data length is %d\n", dataLength);
case CSV:
printf( Intervals One-way Latency for" _dataLen  " Bytes);
printf( "Length (Bytes), Average (us), Std Dev (us),   Min (us),    Max (us),    50% (us),    90% (us),   99% (us),   99.99% (us),  99.9999% (us)");

```

```Length (Bytes), Average (us), Std Dev (us),   Min (us),    Max (us),    50% (us),    90% (us),   99% (us),   99.99% (us),  99.9999% ")```